### ISSUE TITLE
How to delete and query running tasks?

### ISSUE THREAD

**Issue by sunkaifei** [dedefer/tokio_schedule#8]
For example, if this task is launched on the web and all tasks are queried through the database, it involves task deletion. Currently, it seems that there is no task deletion operation in the example.

**Comment by dedefer** [comment]
Hello, [Job.perform](https://docs.rs/tokio_schedule/latest/tokio_schedule/trait.Job.html#method.perform) returns infinite future. The future in rust must be spawned into event loop.

In my examples future is spawned by [`tokio::task::spawn`](https://docs.rs/tokio/latest/tokio/task/fn.spawn.html) that returns JoinHandle. So you can cancel the task by [`JoinHandle::abort`](https://docs.rs/tokio/latest/tokio/task/struct.JoinHandle.html#method.abort).

In other words you can write:

```rs
let weekly = every(1).week().on(Weekday::Mon).at(12, 00, 00)
        .in_timezone(&Utc).perform(|| async { println!("Every week job") });
let weekly_spawned = spawn(weekly);

weekly_spawned.abort() // when you want to drop the job
```

**Comment by sunkaifei** [comment]
Thank you for your reply. How can I ensure that this task can be viewed or added or removed globally during runtime? For example, in the activ web framework, I may need to add or delete tasks on multiple controller pages.

**Comment by dedefer** [comment]
I see, but this library only provides the future objects you can run with tokio. It is not responsible for task management in event loop.
I think for your task you need something [like this](https://github.com/tokio-rs/console).
